src/Utilities/CartesianProduct.hpp

```diff
@@ -0,0 +1,64 @@
+// Distributed under the MIT License.
```

Comment:
This would be good as a separate commit because it is conceptually unrelated to the domain changes.

Reply:
It is a separate commit :)